src/jit/codegenxarch.cpp

```diff
@@ -2768,7 +2778,8 @@ void CodeGen::genCodeForInitBlkRepStos(GenTreeBlk* initBlkNode)
 #ifdef _TARGET_AMD64_
         assert(size > CPBLK_UNROLL_LIMIT && size < CPBLK_MOVS_LIMIT);
 #else
-        assert(size > CPBLK_UNROLL_LIMIT);
+        // Note that a size of zero means a non-constant size.
+        assert((size == 0) || (size > CPBLK_UNROLL_LIMIT));
```

Comment:
This assert has been around for a long time (but in the previous IR implementation this path was only taken if the size was constant). For the x64 case, we only get here if it's constant, but as there's no helper call for x86, we need to handle this case.
If, by some chance, we had constant propagation which caused the size node to be a constant zero, it would still be considered a "non-constant" size, because the size would be a node instead of a constant field on the blk node.